contracts/pools/PoolMigrator.sol

```diff
@@ -136,25 +138,31 @@ contract PoolMigrator is IPoolMigrator, Upgradeable, Utils {
     /**
      * @dev migrates a V1 pool to the latest pool version
      */
-    function _migrateFromV1(
+    function _migrateFromV2(
         Token pool,
-        IPoolCollectionV1 sourcePoolCollection,
+        IPoolCollectionV2 sourcePoolCollection,
         IPoolCollection targetPoolCollection
     ) private {
-        IPoolCollectionV1.PoolV1 memory data = sourcePoolCollection.poolData(pool);
+        IPoolCollectionV2.PoolV2 memory data = sourcePoolCollection.poolData(pool);
+        IPoolCollectionV2.AverageRateV2 memory averageRate = data.averageRate;
+        PoolLiquidity memory liquidity = data.liquidity;
 
-        // since the latest pool collection is also v1, currently not additional pre- or post-processing is needed
+        // since the latest pool collection is also v2, currently not additional pre- or post-processing is needed
         Pool memory newData = Pool({
             poolToken: data.poolToken,
             tradingFeePPM: data.tradingFeePPM,
             tradingEnabled: data.tradingEnabled,
             depositingEnabled: data.depositingEnabled,
-            averageRate: data.averageRate,
-            depositLimit: data.depositLimit,
+            averageRates: AverageRates({
+                blockNumber: averageRate.blockNumber,
+                rate: averageRate.rate,
+                invRate: Fraction({ n: liquidity.baseTokenTradingLiquidity, d: liquidity.bntTradingLiquidity })
```

Comment:
Shouldn't we take the inverse of `averageRate.rate` upon migration?

Reply:
Since we don't actually know the aggregated value of the inverse, I'd figured to start with it to be exactly equal to the rate and let it diverge from there. @yudilevi WDYT?